packages/fdc3-agent-proxy/src/channels/DefaultChannelSupport.ts

```diff
@@ -54,23 +54,24 @@ export class DefaultChannelSupport implements ChannelSupport {
     });
 
     this.addChannelChangedEventHandler(async (e: ApiEvent) => {
-      const cce: ChannelChangedEvent['payload'] = e.details;
-      Logger.debug('Desktop Agent reports channel changed: ', cce.newChannelId);
+      const cce = e as FDC3ChannelChangedEvent;
+      const currentChannelId = cce.details.currentChannelId;
+      Logger.debug('Desktop Agent reports channel changed: ', currentChannelId);
 
       let theChannel: Channel | null = null;
 
       // if theres a newChannelId, retrieve details of the channel
-      if (cce.newChannelId) {
-        theChannel = this.userChannels.find(uc => uc.id == cce.newChannelId) ?? null;
+      if (currentChannelId) {
```

Comment:
it is better to do a more explicit comparison

```ts
if (currentChannelId != null) {
```

in this case if the channel id is `0` or an empty string then the if statement will not fire.

I've had some very long frustrating debug session caused by JavaScript type coercion "feature" where it has not fired an if for a value of `0`